conbench/api/_examples.py

```diff
@@ -354,24 +354,31 @@ def _api_context_entity(context_id, links=True):
 
 
 def _api_history_entity(benchmark_id, case_id, context_id, run_name):
-    return {
-        "benchmark_id": benchmark_id,
-        "case_id": case_id,
-        "context_id": context_id,
-        "hardware_hash": "diana-2-2-4-17179869184",
-        "unit": "s",
-        "data": BENCHMARK_ENTITY["stats"]["data"],
-        "times": BENCHMARK_ENTITY["stats"]["times"],
-        "mean": 0.036369,
-        "begins_distribution_change": False,
-        "distribution_mean": 0.036369,
-        "distribution_stdev": 0.000000,
-        "repository": "https://github.com/org/repo",
-        "sha": "02addad336ba19a654f9c857ede546331be7b631",
-        "timestamp": "2021-02-25T01:02:51",
-        "message": "ARROW-11771: [Developer][Archery] Move benchmark tests (so CI runs them)",
-        "run_name": run_name,
-    }
+    return [
+        {
+            "benchmark_result_id": benchmark_id,
+            "case_id": case_id,
+            "commit_hash": "02addad336ba19a654f9c857ede546331be7b631",
+            "commit_msg": "ARROW-11771: [Developer][Archery] Move benchmark tests (so CI runs them)",
+            "commit_timestamp": "2021-02-25T01:02:51",
+            "context_id": context_id,
+            "data": BENCHMARK_ENTITY["stats"]["data"],
+            "hardware_hash": "diana-2-2-4-17179869184",
+            "mean": 0.036369,
+            "repository": "https://github.com/org/repo",
+            "run_name": run_name,
+            "times": BENCHMARK_ENTITY["stats"]["times"],
+            "unit": "s",
+            "zscorestats": {
+                "begins_distribution_change": False,
+                "residual": 0.0,
+                "rolling_mean": 0.036369,
+                "rolling_mean_excluding_this_commit": 0.036369,
+                "rolling_stddev": 0.0,
+                "segment_id": 0.0,
+            },
+        }
+    ]
```

Comment:
new: always emit JSON array at the outest level, even if it's just one item in it (easier for clients to integrate)

Reply:
I love this change, though we might need to think about updating our clients in various places before we merge this. I can think of a handful of places we use this now within the conbench ecosystem, as well as some of the our reporting infrastructure: https://github.com/conbench/conbenchcoms for example. It shouldn't be too hard to update all of those, but we'll need to coordinate that to make sure we're ready for that to land when it does.

Reply:
Same is true for `sha` -> `commit_hash` I love it, it's the right answer, but it will likely require some work from others (mostly thinking of @boshek but there are others too) to prepare for this change to go live

Reply:
Replied below: https://github.com/conbench/conbench/pull/811#issuecomment-1459771722

Most importantly, I'd appreciate if we can simply list the places where we expect breakage, and I am happy to learn about them and contribute patches. Thanks.